core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/CamelFilesFinder.java

```diff
@@ -69,6 +69,7 @@ private boolean isWorkProjectFolder(IProject project, IResource f) {
 	 */
 	public boolean isFuseCamelContentType(IFile ifile) throws CoreException {
 		return ifile != null
+				&& ifile.isSynchronized(IResource.DEPTH_ZERO)

```

Comment:
I was also wondering for that.
On the other hand, I don't find that this part of the code should be responsible for the refresh. It can mean that there are still actions processing and that the file might change again. This code will be called again after the refresh is finished.